client/client.go

```diff
@@ -552,9 +558,14 @@ func (c *Client) Do(req *http.Request) (*container.Container, *http.Response, er
 			if ok := c.backoff(attempts); !ok {
 				obj, err := container.ParseJSON(bodyBytes)
 				if err != nil {
-					log.Printf("[ERROR] Error occured while json parsing %+v with HTTP StatusCode 405, 500-504", err)
+					log.Printf("[ERROR] Error occured while json parsing: %+v", err)
+
+					// If nginx is too busy or the page is not found, APIC's nginx will response with an HTML doc instead of a JSON Response.
+					// In those cases, parse the HTML response for the message and return that to the user
+					htmlErr := c.checkHtmlResp(bodyStr)
+					log.Printf("[ERROR] Error occured while json parsing: %s", htmlErr.Error())
 					log.Printf("[DEBUG] Exit from Do method")
-					return nil, resp, errors.New(fmt.Sprintf("Failed to parse JSON response from %s. Verify that you are connecting to an APIC.\nHTTP response status: %s\nMessage: %+v", req.URL.String(), resp.Status, err))
+					return nil, resp, errors.New(fmt.Sprintf("API URL: %s, Verify that you are connecting to an APIC.\nHTTP response status: %s\nMessage: %s", req.URL.String(), resp.Status, htmlErr))
```

Comment:
Why change the log error message being returned?

Reply:
the APIC is returning either an JSON body or an HTML body.  if you return this error message, its not actually conveying the error message that is being returned from APIC.  If JSON Parsing fails, that means APIC is returning an HTML Body - so what i'm doing here is extracting the message from the HTML body and returning that as the actual error message.

Reply:
I am only talking about keeping the start of the log message "Failed to parse JSON response from %s." we can keep the extracted HTML body part. Just not sure why we don't keep the start of the log message as "API URL: %s" does not really tell us what is happening either.